rule/add_constant.go

```diff
@@ -36,13 +37,14 @@ type AddConstantRule struct {
 	allowList       allowList
 	ignoreFunctions []*regexp.Regexp
 	strLitLimit     int
-
-	configureOnce sync.Once
 }
 
 // Apply applies the rule to given file.
-func (r *AddConstantRule) Apply(file *lint.File, arguments lint.Arguments) []lint.Failure {
-	r.configureOnce.Do(func() { r.configure(arguments) })
+func (r *AddConstantRule) Apply(file *lint.File, arguments lint.Arguments) ([]lint.Failure, error) {
+	check := sync.OnceValue(func() error {return r.configure(arguments)})
+	if err := check(); err != nil {
+		return nil, err
+	}
```

Comment:
This won't work as you expected. The `configure` will be called multiple times.

You can check this by adding the statement `fmt.Println("configure add-constant")` inside the `func() error`. After running `revive` with enabled `add-constant` on a real project with multiple packages, there are a lot of lines `configure add-constant`.

Reply:
ok iam back to concept with local error variable:

```
var configureErr error                                                                                                                                                 
 r.configureOnce.Do(func() { configureErr = r.configure(arguments) })                                                                                                   
                                                                                                                                                                           
    if configureErr != nil {                                                                                                                                               
      return nil, configureErr                                                                                                                                           
    }
```
but now executing ``` r.configureOnce.Do() ``` runs only onece (added ``` fmt.Println("configure add-constant")``` to configure() and run on large project), and on output we have only one message on stdout